### PR TITLE
[MWRAPPER-104] add prettier config for shell

### PR DESCRIPTION
**Pull request by bmarwell**
This is an alternate proposal for running a bash linter. This time it integrates with spotless (and you can use spotlesss:apply). The other approach is using shellcheckk (https://github.com/apache/maven-wrapper/pull/82) which is a little more known and does not require npm modules, but does not integrate with spotless as nicely.

Signed-off-by: Benjamin Marwell <bmarwell@apache.org>

Fixes MWRAPPER-104

---

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [X] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MWRAPPER-104) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [X] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MWRAPPER-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MWRAPPER-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [ ] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [X] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [X] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by bmarwell** [comment]
You might want to talk about:
https://github.com/apache/maven-wrapper/pull/84/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R110-R111

`onlymvnw` uses a few places with spaces (`  case)` and `> &2`) wheras `mvnw` uses `case)` and `>&2`. So both will work for me, we just would need to decide.

**Review by slawekjaranowski — COMMENTED** [review]
Looks ok for me

**Comment by slawekjaranowski** [comment]
By the way issue will be appreciated for it.

**Review by slawekjaranowski — APPROVED** [review]
Please apply new format to make a build pass 😄

**Review by slachiewicz — APPROVED** [review]

**Comment by bmarwell** [comment]
@slachiewicz @slawekjaranowski rebased and added a commit which applies the prettier config

**Comment by slawekjaranowski** [comment]
@bmarwell  thanks ... I did forgot it 😄 

I see that we should first update parent to the latest in order to build under JDK 21

**Comment by bmarwell** [comment]
> I see that we should first update parent to the latest in order to build under JDK 21

Was about to say the same. Yup. There are a couple more  PRs we should put into the next release.

**Review by slachiewicz — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -77,6 +77,44 @@ under the License.
           <tagNameFormat>maven-wrapper-@{project.version}</tagNameFormat>
         </configuration>
       </plugin>
+
+      <plugin>
+        <groupId>com.diffplug.spotless</groupId>
+        <artifactId>spotless-maven-plugin</artifactId>
```
Comment (slawekjaranowski) [inline pom.xml:83]: Please move to pluginManagement

**Comment by jira-importer** [comment]
Resolve #288

**Comment by jira-importer** [comment]
Resolve #288

**Pull request merged**